File: frontend/src/types/notification.ts
export interface Notification {
  id?: number;
  type: 'connection' | 'new_tweet' | 'new_like' | 'new_retweet' | 'new_reply' | 'new_follower' | 'new_message';
  message: string;
  related_id?: number;
  related_username?: string;
  is_read?: boolean;
  created_at?: string;
  data?: any;
  timestamp: string;
}

File: backend/app/models/notification.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Boolean, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.base import Base

class Notification(Base):
    __tablename__ = "notifications"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    type = Column(String, nullable=False)  # new_like, new_reply, new_follower, new_message
    message = Column(Text, nullable=False)
    related_id = Column(Integer)  # ID del tweet, usuario, etc relacionado
    related_username = Column(String)  # Username relacionado para navegación
    is_read = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # Relaciones
    user = relationship("User", backref="notifications")